pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -81,7 +81,7 @@ func (b *AccessLogBuilder) setTCPAccessLog(push *model.PushContext, proxy *model
 	mesh := push.Mesh
 	cfgs := push.Telemetry.AccessLogging(push, proxy, class)
 
-	if cfgs == nil {
+	if len(cfgs) == 0 {
```

Comment:
why need this?
`nil` means there's no `telemetry api`

Reply:
Fix this case: https://github.com/istio/istio/blob/7d0061176b1cd4383f4c9ef99d2db95291063796/pilot/pkg/networking/core/v1alpha3/accesslog_test.go#L459

described here: https://github.com/istio/istio/issues/43371#:~:text=Example%2C-,case%201,-%3A%20this%20filter%20takes

Reply:
can you retest without this changes?

Reply:
<img width="1919" alt="image" src="https://user-images.githubusercontent.com/7346661/220042371-0553b74f-cc51-4317-8eef-bc7528d3a659.png">

Reply:
I got this point, this's someting was talked in https://github.com/istio/istio/pull/39521#discussion_r941833914.

thanks for catching this!

Reply:
@zirain I found that there will be a new problem here, which will cause the disable access logs to fail. However, I am confused as to why this e2e test passed. https://github.com/istio/istio/blob/master/tests/integration/telemetry/api/accesslogs_test.go#L58

It failed when I try to cherry-pick it to release 1.16.

Reply:
@zirain I have submitted a pull request to fix this issue: #43770